# ML Inital Exploration 

A early approach to ML with the FIRMS+SCAN data mergring with the "larger" WFIGS dataset

This file is missing the "Number of nearby detections" field, which we will have to run later

### Approach:
Sections of this notebook

#### Get Data and Merge 
- Started by merging FIRMS+SCAN file ('FIRMSandSCANFull2018toApr2022.csv') from the team's S3 bucket
- That dataframe was then merged with WFIGS actual fire database 

#### Clean And Prep the Data
- Remove unnessary columns 
- Remove columns we dont want to inclide for this inital approach  
- Remove rows with nan
- One hot encode necessary columns

#### ML preperation
- divide X and y for features and labels
- standardize the data

#### ML Running and Evaluation 
- Random Forest Classifier 200 estimators
- Random Forest Classifier 100 estimators
- Neural Network
- Logistic Regression
- Logistic regression CV
- SGD Classifier
- Extra Trees Classifier
- Bagging Classifier


In [1]:
#general imports
import pandas as pd
import boto3
import geopy.distance  
import numpy as np
pd.set_option('display.max_columns', 200)

## Load and Merge The Data

In [2]:
#load in the csvs
#TODO For Team: enter the credentails below to run
S3_Key_id=''
S3_Secret_key=''

def pull_data(Key_id, Secret_key, file):
    """
    Function which CJ wrote to pull data from S3 
    """
    BUCKET_NAME = "gtown-wildfire-ds"
    OBJECT_KEY = file
    client = boto3.client(
        's3',
        aws_access_key_id= Key_id,
        aws_secret_access_key= Secret_key)
    obj = client.get_object(Bucket= BUCKET_NAME, Key= OBJECT_KEY) 
    file_df = pd.read_csv(obj['Body'])
    return (file_df)

#Pull in the firms and scan df
file = 'FIRMSandSCANFull2018toApr2022.csv'
df_FirmsAndScan = pull_data(S3_Key_id, S3_Secret_key, file)
df_FirmsAndScan.head()

C:\Users\anderb4\AppData\Local\Temp\ipykernel_10896\1585433451.py:17: DtypeWarning: Columns (8,10,11) have mixed types. Specify dtype option on import or set low_memory=False.
  file_df = pd.read_csv(obj['Body'])


Unnamed: 0  latitude  longitude  brightness  scan  track    acq_date  \
0           0   33.4856   -90.2127       319.4   1.4    1.2  2018-01-01   
1           1   44.3762  -119.1190       312.5   1.2    1.1  2018-01-01   
2           2   19.4122  -155.2699       309.6   3.9    1.8  2018-01-01   
3           3   19.3431  -155.0508       309.8   4.0    1.9  2018-01-01   
4           4   19.3853  -155.0768       312.8   4.0    1.9  2018-01-01   

   acq_time satellite instrument confidence version  bright_t31    frp  \
0       427     Terra      MODIS         98    6.03       267.5   38.3   
1       608     Terra      MODIS         85    6.03       269.1   21.9   
2       919     Terra      MODIS         67    6.03       284.0   90.7   
3       919     Terra      MODIS         68    6.03       289.9   82.0   
4       919     Terra      MODIS         80    6.03       287.1  103.7   

  daynight  type   lat   long  hour                date_loc  \
0        N   0.0  33.5  -90.2   4.0   2018-01-01,33.5,-90.2   
1        N   0.0  44.4 -119.1   6.0  2018-01-01,44.4,-119.1   
2        N   1.0  19.4 -155.3   9.0  2018-01-01,19.4,-155.3   
3        N   1.0  19.3 -155.1   9.0  2018-01-01,19.3,-155.1   
4        N   1.0  19.4 -155.1   9.0  2018-01-01,19.4,-155.1   

               date_hour_loc     newdate       nearestStation  StationDist  \
0   2018-01-01,04,33.5,-90.2  2018-01-01  Shaw Nature Reserve    38.496839   
1  2018-01-01,06,44.4,-119.1  2018-01-01       Molly Caren #1   124.725193   
2  2018-01-01,09,19.4,-155.3  2018-01-01  Shaw Nature Reserve    39.500776   
3  2018-01-01,09,19.3,-155.1  2018-01-01             Mason #1   170.344414   
4  2018-01-01,09,19.4,-155.1  2018-01-01           Uapb-Earle    56.557018   

         Date  Station Id State Code Network Code    Station Name  \
0         NaN         NaN        NaN          NaN             NaN   
1  2018-01-01      2014.0         OH         SCAN  Molly Caren #1   
2         NaN         NaN        NaN          NaN             NaN   
3  2018-01-01      2004.0         IL         SCAN        Mason #1   
4  2018-01-01      2085.0         AR         SCAN      Uapb-Earle   

   Elevation (ft)  Latitude  Longitude County Name                 HUC2 Name  \
0             NaN       NaN        NaN         NaN                       NaN   
1          1060.0  39.95000  -83.45000     Madison               Ohio Region   
2             NaN       NaN        NaN         NaN                       NaN   
3           570.0  40.31667  -89.90000       Mason  Upper Mississippi Region   
4           213.0  35.27732  -90.45031  Crittenden  Lower Mississippi Region   

                       HUC4 Name       HUC6 Name  \
0                            NaN             NaN   
1                         Scioto          Scioto   
2                            NaN             NaN   
3                 Lower Illinois  Lower Illinois   
4  Lower Mississippi-St. Francis     St. Francis   

                        HUC8 Name  \
0                             NaN   
1                    Lower Scioto   
2                             NaN   
3  Lower Illinois-Lake Chautauqua   
4               Lower St. Francis   

   Precipitation Accumulation (in) Start of Day Values  \
0                                                NaN     
1                                               11.5     
2                                                NaN     
3                                                7.8     
4                                               11.3     

   Precipitation Increment (in)  \
0                           NaN   
1                           0.0   
2                           NaN   
3                           0.0   
4                           0.0   

   Precipitation Month-to-date (in) Start of Day Values  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                              

In [3]:
#pull in the big WFIGS dataframe
file = 'WFIGS_big_Pulled5-8-2022.csv'
df_wfigsbig = pull_data(S3_Key_id, S3_Secret_key, file)
df_wfigsbig.head()

C:\Users\anderb4\AppData\Local\Temp\ipykernel_10896\1585433451.py:17: DtypeWarning: Columns (13,14,32,61,64,94,95) have mixed types. Specify dtype option on import or set low_memory=False.
  file_df = pd.read_csv(obj['Body'])


X          Y  OBJECTID ABCDMisc ADSPermissionState  \
0 -104.457511  45.785037         1      NaN          CERTIFIED   
1 -118.180712  33.808985         2      NaN            DEFAULT   
2 -114.835414  48.073946         3      NaN            DEFAULT   
3 -117.153901  33.176394         4      NaN            DEFAULT   
4 -121.104180  38.834727         5      NaN            DEFAULT   

   CalculatedAcres     ContainmentDateTime         ControlDateTime  \
0            50.64  2020/08/06 23:13:07+00  2020/08/06 23:13:24+00   
1              NaN                     NaN                     NaN   
2              NaN  2017/10/18 00:30:00+00  2017/10/18 00:35:00+00   
3              NaN                     NaN                     NaN   
4              NaN                     NaN                     NaN   

   DailyAcres  DiscoveryAcres DispatchCenterID  EstimatedCostToDate  \
0        50.6            20.0            MTMCC                  NaN   
1         NaN             0.1           CALACC                  NaN   
2        50.0            50.0            MTKIC                  NaN   
3         NaN             NaN           CAMVIC                  NaN   
4         NaN             NaN              NaN                  NaN   

   FinalFireReportApprovedByTitle FinalFireReportApprovedByUnit  \
0                             NaN                           NaN   
1                             NaN                           NaN   
2                             NaN                           NaN   
3                             NaN                           NaN   
4                             NaN                           NaN   

  FinalFireReportApprovedDate FireBehaviorGeneral FireBehaviorGeneral1  \
0      2020/08/12 20:45:59+00                 NaN                  NaN   
1                         NaN                 NaN                  NaN   
2                         NaN                 NaN                  NaN   
3                         NaN                 NaN                  NaN   
4                         NaN                 NaN                  NaN   

  FireBehaviorGeneral2 FireBehaviorGeneral3 FireCause FireCauseGeneral  \
0                  NaN                  NaN     Human        Equipment   
1                  NaN                  NaN   Unknown              NaN   
2                  NaN                  NaN     Human              NaN   
3                  NaN                  NaN       NaN              NaN   
4                  NaN                  NaN       NaN              NaN   

   FireCauseSpecific FireCode FireDepartmentID   FireDiscoveryDateTime  \
0  Farming Equipment     NDU0              NaN  2020/08/06 18:58:00+00   
1                NaN      NaN              NaN  2020/02/28 20:45:40+00   
2                NaN      NaN              NaN  2017/10/17 20:20:24+00   
3                NaN     MC2S              NaN  2019/07/01 19:54:00+00   
4                NaN     KA82              NaN  2016/06/20 22:05:59+00   

  FireMgmtComplexity         FireOutDateTime  FireStrategyConfinePercent  \
0                NaN  2020/08/12 14:00:00+00                         NaN   
1                NaN                     NaN                         NaN   
2                NaN  2017/11/09 21:59:59+00                         NaN   
3                NaN                     NaN                         NaN   
4                NaN                     NaN                         NaN   

   FireStrategyFullSuppPercent  FireStrategyMonitorPercent  \
0                          NaN                         NaN   
1                          NaN                         NaN   
2                          NaN                         NaN   
3                          NaN                         NaN   
4                          NaN                         NaN   

   FireStrategyPointZonePercent FSJobCode FSOverrideCode  GACC  \
0                           NaN        PN           1502  NRCC   
1                           NaN       NaN            NaN  OSCC   
2

In [4]:
# now starting to transform the fire db
latlonground = 1 #Sets the number of decimal places to round latlong
#getting the fire discovered date into the same format as the same date
df_wfigsbig['disc_date'] = df_wfigsbig['FireDiscoveryDateTime'].astype(str).str[0:10]
df_wfigsbig['disc_date'] = df_wfigsbig['disc_date'].astype(str).str.replace('/', '-')
# #convert the init long and lat frame to 2 decimals like i did in the satellite df 
df_wfigsbig['init_lat_rounded'] = round(df_wfigsbig['Y'], latlonground) #rounds lat long into new column, using latlonground # of decimal pts
df_wfigsbig['init_long_rounded'] = round(df_wfigsbig['X'], latlonground) #rounds lat long into new column, using latlonground # of decimal pts
#make the date loc column
df_wfigsbig['disc_date_loc'] = df_wfigsbig['disc_date'] + ',' + df_wfigsbig['init_lat_rounded'].astype(str) + ',' + df_wfigsbig['init_long_rounded'].astype(str)
df_wfigsbig.head()

X          Y  OBJECTID ABCDMisc ADSPermissionState  \
0 -104.457511  45.785037         1      NaN          CERTIFIED   
1 -118.180712  33.808985         2      NaN            DEFAULT   
2 -114.835414  48.073946         3      NaN            DEFAULT   
3 -117.153901  33.176394         4      NaN            DEFAULT   
4 -121.104180  38.834727         5      NaN            DEFAULT   

   CalculatedAcres     ContainmentDateTime         ControlDateTime  \
0            50.64  2020/08/06 23:13:07+00  2020/08/06 23:13:24+00   
1              NaN                     NaN                     NaN   
2              NaN  2017/10/18 00:30:00+00  2017/10/18 00:35:00+00   
3              NaN                     NaN                     NaN   
4              NaN                     NaN                     NaN   

   DailyAcres  DiscoveryAcres DispatchCenterID  EstimatedCostToDate  \
0        50.6            20.0            MTMCC                  NaN   
1         NaN             0.1           CALACC                  NaN   
2        50.0            50.0            MTKIC                  NaN   
3         NaN             NaN           CAMVIC                  NaN   
4         NaN             NaN              NaN                  NaN   

   FinalFireReportApprovedByTitle FinalFireReportApprovedByUnit  \
0                             NaN                           NaN   
1                             NaN                           NaN   
2                             NaN                           NaN   
3                             NaN                           NaN   
4                             NaN                           NaN   

  FinalFireReportApprovedDate FireBehaviorGeneral FireBehaviorGeneral1  \
0      2020/08/12 20:45:59+00                 NaN                  NaN   
1                         NaN                 NaN                  NaN   
2                         NaN                 NaN                  NaN   
3                         NaN                 NaN                  NaN   
4                         NaN                 NaN                  NaN   

  FireBehaviorGeneral2 FireBehaviorGeneral3 FireCause FireCauseGeneral  \
0                  NaN                  NaN     Human        Equipment   
1                  NaN                  NaN   Unknown              NaN   
2                  NaN                  NaN     Human              NaN   
3                  NaN                  NaN       NaN              NaN   
4                  NaN                  NaN       NaN              NaN   

   FireCauseSpecific FireCode FireDepartmentID   FireDiscoveryDateTime  \
0  Farming Equipment     NDU0              NaN  2020/08/06 18:58:00+00   
1                NaN      NaN              NaN  2020/02/28 20:45:40+00   
2                NaN      NaN              NaN  2017/10/17 20:20:24+00   
3                NaN     MC2S              NaN  2019/07/01 19:54:00+00   
4                NaN     KA82              NaN  2016/06/20 22:05:59+00   

  FireMgmtComplexity         FireOutDateTime  FireStrategyConfinePercent  \
0                NaN  2020/08/12 14:00:00+00                         NaN   
1                NaN                     NaN                         NaN   
2                NaN  2017/11/09 21:59:59+00                         NaN   
3                NaN                     NaN                         NaN   
4                NaN                     NaN                         NaN   

   FireStrategyFullSuppPercent  FireStrategyMonitorPercent  \
0                          NaN                         NaN   
1                          NaN                         NaN   
2                          NaN                         NaN   
3                          NaN                         NaN   
4                          NaN                         NaN   

   FireStrategyPointZonePercent FSJobCode FSOverrideCode  GACC  \
0                           NaN        PN           1502  NRCC   
1                           NaN       NaN            NaN  OSCC   
2

In [5]:
#Merging on the date + location key.
dfDiscovered = df_FirmsAndScan.merge(df_wfigsbig, left_on='date_loc', right_on='disc_date_loc', how='inner') #creates df of only dsetected areas
dfMerged = df_FirmsAndScan.merge(df_wfigsbig, left_on='date_loc', right_on='disc_date_loc', how='left') #creates df of both 
dfMerged['FIRE_DETECTED'] = dfMerged['OBJECTID'].isnull() #if there was no fire detectection set to true
dfMerged['FIRE_DETECTED'] = ~dfMerged['FIRE_DETECTED'].astype(bool) #flip so that if detection merged set to true
print(dfMerged.shape)
dfMerged.head()

(2096070, 149)


Unnamed: 0  latitude  longitude  brightness  scan  track    acq_date  \
0           0   33.4856   -90.2127       319.4   1.4    1.2  2018-01-01   
1           1   44.3762  -119.1190       312.5   1.2    1.1  2018-01-01   
2           2   19.4122  -155.2699       309.6   3.9    1.8  2018-01-01   
3           3   19.3431  -155.0508       309.8   4.0    1.9  2018-01-01   
4           4   19.3853  -155.0768       312.8   4.0    1.9  2018-01-01   

   acq_time satellite instrument confidence version  bright_t31    frp  \
0       427     Terra      MODIS         98    6.03       267.5   38.3   
1       608     Terra      MODIS         85    6.03       269.1   21.9   
2       919     Terra      MODIS         67    6.03       284.0   90.7   
3       919     Terra      MODIS         68    6.03       289.9   82.0   
4       919     Terra      MODIS         80    6.03       287.1  103.7   

  daynight  type   lat   long  hour                date_loc  \
0        N   0.0  33.5  -90.2   4.0   2018-01-01,33.5,-90.2   
1        N   0.0  44.4 -119.1   6.0  2018-01-01,44.4,-119.1   
2        N   1.0  19.4 -155.3   9.0  2018-01-01,19.4,-155.3   
3        N   1.0  19.3 -155.1   9.0  2018-01-01,19.3,-155.1   
4        N   1.0  19.4 -155.1   9.0  2018-01-01,19.4,-155.1   

               date_hour_loc     newdate       nearestStation  StationDist  \
0   2018-01-01,04,33.5,-90.2  2018-01-01  Shaw Nature Reserve    38.496839   
1  2018-01-01,06,44.4,-119.1  2018-01-01       Molly Caren #1   124.725193   
2  2018-01-01,09,19.4,-155.3  2018-01-01  Shaw Nature Reserve    39.500776   
3  2018-01-01,09,19.3,-155.1  2018-01-01             Mason #1   170.344414   
4  2018-01-01,09,19.4,-155.1  2018-01-01           Uapb-Earle    56.557018   

         Date  Station Id State Code Network Code    Station Name  \
0         NaN         NaN        NaN          NaN             NaN   
1  2018-01-01      2014.0         OH         SCAN  Molly Caren #1   
2         NaN         NaN        NaN          NaN             NaN   
3  2018-01-01      2004.0         IL         SCAN        Mason #1   
4  2018-01-01      2085.0         AR         SCAN      Uapb-Earle   

   Elevation (ft)  Latitude  Longitude County Name                 HUC2 Name  \
0             NaN       NaN        NaN         NaN                       NaN   
1          1060.0  39.95000  -83.45000     Madison               Ohio Region   
2             NaN       NaN        NaN         NaN                       NaN   
3           570.0  40.31667  -89.90000       Mason  Upper Mississippi Region   
4           213.0  35.27732  -90.45031  Crittenden  Lower Mississippi Region   

                       HUC4 Name       HUC6 Name  \
0                            NaN             NaN   
1                         Scioto          Scioto   
2                            NaN             NaN   
3                 Lower Illinois  Lower Illinois   
4  Lower Mississippi-St. Francis     St. Francis   

                        HUC8 Name  \
0                             NaN   
1                    Lower Scioto   
2                             NaN   
3  Lower Illinois-Lake Chautauqua   
4               Lower St. Francis   

   Precipitation Accumulation (in) Start of Day Values  \
0                                                NaN     
1                                               11.5     
2                                                NaN     
3                                                7.8     
4                                               11.3     

   Precipitation Increment (in)  \
0                           NaN   
1                           0.0   
2                           NaN   
3                           0.0   
4                           0.0   

   Precipitation Month-to-date (in) Start of Day Values  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                              

In [6]:
#check the number of merges
dfDiscovered.shape

(63846, 148)

## Clean And Prep the Data

In [7]:
#beign cleaning the data by removing unnecessary columns:
#drop the columns we VERY likely do not need for this approach
coltodrop = ['Unnamed: 0',
'acq_date',
'acq_time',
'version',
'lat',
'long',
'hour',
'date_loc',
'date_hour_loc',
'newdate',
'nearestStation',
'Date',
'Station Id',
'State Code',
'Network Code',
'Station Name',
'Elevation (ft)',
'Latitude',
'Longitude',
'County Name',
'HUC6 Name',
'HUC8 Name',
'Wind Speed Average 324in (mph)',
'Wind Speed Average 128in (mph)',
'date_new',
'X',
'Y',
'OBJECTID',
'ABCDMisc',
'ADSPermissionState',
'CalculatedAcres',
'ContainmentDateTime',
'ControlDateTime',
'DailyAcres',
'DiscoveryAcres',
'DispatchCenterID',
'EstimatedCostToDate',
'FinalFireReportApprovedByTitle',
'FinalFireReportApprovedByUnit',
'FinalFireReportApprovedDate',
'FireBehaviorGeneral',
'FireBehaviorGeneral1',
'FireBehaviorGeneral2',
'FireBehaviorGeneral3',
'FireCause',
'FireCauseGeneral',
'FireCauseSpecific',
'FireCode',
'FireDepartmentID',
'FireDiscoveryDateTime',
'FireMgmtComplexity',
'FireOutDateTime',
'FireStrategyConfinePercent',
'FireStrategyFullSuppPercent',
'FireStrategyMonitorPercent',
'FireStrategyPointZonePercent',
'FSJobCode',
'FSOverrideCode',
'GACC',
'ICS209ReportDateTime',
'ICS209ReportForTimePeriodFrom',
'ICS209ReportForTimePeriodTo',
'ICS209ReportStatus',
'IncidentManagementOrganization',
'IncidentName',
'IncidentShortDescription',
'IncidentTypeCategory',
'IncidentTypeKind',
'InitialLatitude',
'InitialLongitude',
'InitialResponseAcres',
'InitialResponseDateTime',
'IrwinID',
'IsFireCauseInvestigated',
'IsFireCodeRequested',
'IsFSAssisted',
'IsMultiJurisdictional',
'IsReimbursable',
'IsTrespass',
'IsUnifiedCommand',
'LocalIncidentIdentifier',
'PercentContained',
'PercentPerimeterToBeContained',
'POOCity',
'POOCounty',
'POODispatchCenterID',
'POOFips',
'POOJurisdictionalAgency',
'POOJurisdictionalUnit',
'POOJurisdictionalUnitParentUnit',
'POOLandownerCategory',
'POOLandownerKind',
'POOLegalDescPrincipalMeridian',
'POOLegalDescQtr',
'POOLegalDescQtrQtr',
'POOLegalDescRange',
'POOLegalDescSection',
'POOLegalDescTownship',
'POOPredictiveServiceAreaID',
'POOProtectingAgency',
'POOProtectingUnit',
'POOState',
'PredominantFuelGroup',
'PredominantFuelModel',
'PrimaryFuelModel',
'SecondaryFuelModel',
'TotalIncidentPersonnel',
'UniqueFireIdentifier',
'WFDSSDecisionStatus',
'CreatedBySystem',
'ModifiedBySystem',
'IsDispatchComplete',
'OrganizationalAssessment',
'StrategicDecisionPublishDate',
'CreatedOnDateTime_dt',
'ModifiedOnDateTime_dt',
'Source',
'GlobalID',
'IsCpxChild',
'CpxName',
'CpxID',
'disc_date',
'init_lat_rounded',
'init_long_rounded',
'disc_date_loc'
            ]
dfMerged = dfMerged.drop(coltodrop, axis = 1) #drops unwanted columns for ML process
dfMerged.head()

latitude  longitude  brightness  scan  track satellite instrument  \
0   33.4856   -90.2127       319.4   1.4    1.2     Terra      MODIS   
1   44.3762  -119.1190       312.5   1.2    1.1     Terra      MODIS   
2   19.4122  -155.2699       309.6   3.9    1.8     Terra      MODIS   
3   19.3431  -155.0508       309.8   4.0    1.9     Terra      MODIS   
4   19.3853  -155.0768       312.8   4.0    1.9     Terra      MODIS   

  confidence  bright_t31    frp daynight  type  StationDist  \
0         98       267.5   38.3        N   0.0    38.496839   
1         85       269.1   21.9        N   0.0   124.725193   
2         67       284.0   90.7        N   1.0    39.500776   
3         68       289.9   82.0        N   1.0   170.344414   
4         80       287.1  103.7        N   1.0    56.557018   

                  HUC2 Name                      HUC4 Name  \
0                       NaN                            NaN   
1               Ohio Region                         Scioto   
2                       NaN                            NaN   
3  Upper Mississippi Region                 Lower Illinois   
4  Lower Mississippi Region  Lower Mississippi-St. Francis   

   Precipitation Accumulation (in) Start of Day Values  \
0                                                NaN     
1                                               11.5     
2                                                NaN     
3                                                7.8     
4                                               11.3     

   Precipitation Increment (in)  \
0                           NaN   
1                           0.0   
2                           NaN   
3                           0.0   
4                           0.0   

   Precipitation Month-to-date (in) Start of Day Values  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                                                NaN      
4                                                7.6      

   Air Temperature Average (degF)  \
0                             NaN   
1                             2.0   
2                             NaN   
3                            -8.0   
4                            14.0   

   Soil Moisture Percent -2in (pct) Start of Day Values  \
0                                                NaN      
1                                               36.1      
2                                                NaN      
3                                                4.9      
4                                               26.9      

   Soil Moisture Percent -4in (pct) Start of Day Values  \
0                                                NaN      
1                                               36.4      
2                                                NaN      
3                                                6.2      
4                                               34.0      

   Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  FIRE_DETECTED  
0                                NaN                       NaN          False  
1                               78.0                       7.1          False  
2                                NaN                       NaN          False  
3                               66.0                       1.8          False  
4                               79.0                      10.5          False

In [8]:
#remove columns we (likely) dont need for this approach 
coltodrop = ['latitude',
                'longitude',
                'daynight',
                'Soil Moisture Percent -4in (pct) Start of Day Values',
                'HUC4 Name']
dfMerged = dfMerged.drop(coltodrop, axis = 1) #drops unwanted columns for ML process
dfMerged.head()

brightness  scan  track satellite instrument confidence  bright_t31    frp  \
0       319.4   1.4    1.2     Terra      MODIS         98       267.5   38.3   
1       312.5   1.2    1.1     Terra      MODIS         85       269.1   21.9   
2       309.6   3.9    1.8     Terra      MODIS         67       284.0   90.7   
3       309.8   4.0    1.9     Terra      MODIS         68       289.9   82.0   
4       312.8   4.0    1.9     Terra      MODIS         80       287.1  103.7   

   type  StationDist                 HUC2 Name  \
0   0.0    38.496839                       NaN   
1   0.0   124.725193               Ohio Region   
2   1.0    39.500776                       NaN   
3   1.0   170.344414  Upper Mississippi Region   
4   1.0    56.557018  Lower Mississippi Region   

   Precipitation Accumulation (in) Start of Day Values  \
0                                                NaN     
1                                               11.5     
2                                                NaN     
3                                                7.8     
4                                               11.3     

   Precipitation Increment (in)  \
0                           NaN   
1                           0.0   
2                           NaN   
3                           0.0   
4                           0.0   

   Precipitation Month-to-date (in) Start of Day Values  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                                                NaN      
4                                                7.6      

   Air Temperature Average (degF)  \
0                             NaN   
1                             2.0   
2                             NaN   
3                            -8.0   
4                            14.0   

   Soil Moisture Percent -2in (pct) Start of Day Values  \
0                                                NaN      
1                                               36.1      
2                                                NaN      
3                                                4.9      
4                                               26.9      

   Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  FIRE_DETECTED  
0                                NaN                       NaN          False  
1                               78.0                       7.1          False  
2                                NaN                       NaN          False  
3                               66.0                       1.8          False  
4                               79.0                      10.5          False

In [9]:
#checking some countsd of values on missing columns 
collist = list(dfMerged.columns) #get the list of columns in the df
numbrows = dfMerged.shape[0] #saves number of riows in the df
for col in collist:
    missingcount = dfMerged[col].isnull().sum() #count missing vals 
    missingpercent = round((missingcount/numbrows)*100, 2) #gets the percent compared to the entire dataframe
    print('number of nan for '+col+':'+str(missingcount)+' ; '+str(missingpercent)+'%') #prints the result

number of nan for brightness:0 ; 0.0%
number of nan for scan:0 ; 0.0%
number of nan for track:0 ; 0.0%
number of nan for satellite:0 ; 0.0%
number of nan for instrument:0 ; 0.0%
number of nan for confidence:0 ; 0.0%
number of nan for bright_t31:0 ; 0.0%
number of nan for frp:0 ; 0.0%
number of nan for type:1542889 ; 73.61%
number of nan for StationDist:0 ; 0.0%
number of nan for HUC2 Name:51886 ; 2.48%
number of nan for Precipitation Accumulation (in) Start of Day Values:479812 ; 22.89%
number of nan for Precipitation Increment (in):482593 ; 23.02%
number of nan for Precipitation Month-to-date (in) Start of Day Values:816760 ; 38.97%
number of nan for Air Temperature Average (degF):231959 ; 11.07%
number of nan for Soil Moisture Percent -2in (pct) Start of Day Values:400189 ; 19.09%
number of nan for Relative Humidity Enclosure (pct):185163 ; 8.83%
number of nan for Wind Speed Average (mph):185817 ; 8.87%
number of nan for FIRE_DETECTED:0 ; 0.0%


In [10]:
#there appears to be a lot of missing data for 'type' and 'precipitation month to date'. drop those for this approach
dfMerged = dfMerged.drop('type', axis = 1)
dfMerged = dfMerged.drop('Precipitation Month-to-date (in) Start of Day Values', axis = 1)
dfMerged.head()

brightness  scan  track satellite instrument confidence  bright_t31    frp  \
0       319.4   1.4    1.2     Terra      MODIS         98       267.5   38.3   
1       312.5   1.2    1.1     Terra      MODIS         85       269.1   21.9   
2       309.6   3.9    1.8     Terra      MODIS         67       284.0   90.7   
3       309.8   4.0    1.9     Terra      MODIS         68       289.9   82.0   
4       312.8   4.0    1.9     Terra      MODIS         80       287.1  103.7   

   StationDist                 HUC2 Name  \
0    38.496839                       NaN   
1   124.725193               Ohio Region   
2    39.500776                       NaN   
3   170.344414  Upper Mississippi Region   
4    56.557018  Lower Mississippi Region   

   Precipitation Accumulation (in) Start of Day Values  \
0                                                NaN     
1                                               11.5     
2                                                NaN     
3                                                7.8     
4                                               11.3     

   Precipitation Increment (in)  Air Temperature Average (degF)  \
0                           NaN                             NaN   
1                           0.0                             2.0   
2                           NaN                             NaN   
3                           0.0                            -8.0   
4                           0.0                            14.0   

   Soil Moisture Percent -2in (pct) Start of Day Values  \
0                                                NaN      
1                                               36.1      
2                                                NaN      
3                                                4.9      
4                                               26.9      

   Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  FIRE_DETECTED  
0                                NaN                       NaN          False  
1                               78.0                       7.1          False  
2                                NaN                       NaN          False  
3                               66.0                       1.8          False  
4                               79.0                      10.5          False

In [11]:
#the station distance is not relevant for a fire prediction so dropping
#we may want to drop rows where the nearest station distance is too far in the future
dfMerged = dfMerged.drop('StationDist', axis = 1)

In [12]:
#now drop the nan rows 
print('df shape before dropping all nan: '+str(dfMerged.shape))
dfMerged = dfMerged.dropna() #drop all nan values 
print('df shape after dropping all nan: '+str(dfMerged.shape))

df shape before dropping all nan: (2096070, 16)
df shape after dropping all nan: (1396691, 16)


In [13]:
#check how many trues we have
dfMerged['FIRE_DETECTED'].value_counts()

False    1357647
True       39044
Name: FIRE_DETECTED, dtype: int64

In [14]:
39044/1357647

0.02875858010219151

looks like 2.88% of satellite observation points merged to a actual WFIGS fire, which is believable and okay

In [15]:
#now begin to one hot encode the 'satellite' column
#first check what kind of values we have for the satellite column
dfMerged['satellite'].value_counts()

1        965004
Aqua     217886
Terra    207087
1          6714
Name: satellite, dtype: int64

In [16]:
#weird how '1' appears twice, convert to list to check why and if different datatype
dfMerged['satellite'].unique().tolist()

['Terra', 'Aqua', '1', 1]

In [17]:
#looks like some 1's are strings and some are ints. converting everything to string
dfMerged['satellite'] = dfMerged['satellite'].astype(str)

In [18]:
#check to make sure we are good now
dfMerged['satellite'].value_counts()

1        971718
Aqua     217886
Terra    207087
Name: satellite, dtype: int64

In [19]:
#looks like we are now aligned with the '1''s, now one hot encoding the satellite
#check a one hot encoded satellite df
sat_dummies = pd.get_dummies(dfMerged.satellite)
sat_dummies

1  Aqua  Terra
1        0     0      1
3        0     0      1
4        0     0      1
5        0     0      1
10       0     1      0
...     ..   ...    ...
2096064  1     0      0
2096065  1     0      0
2096066  1     0      0
2096068  1     0      0
2096069  1     0      0

[1396691 rows x 3 columns]

In [20]:
#add the one hot encoded df to the merged dataframe and drop the original satellite column
dfMerged = pd.concat([dfMerged, sat_dummies], axis=1) #adds in the one hot encoded  staellite df
dfMerged = dfMerged.drop('satellite', axis = 1) #drop the satellite column
dfMerged.head()

brightness  scan  track instrument confidence  bright_t31    frp  \
1        312.5   1.2    1.1      MODIS         85       269.1   21.9   
3        309.8   4.0    1.9      MODIS         68       289.9   82.0   
4        312.8   4.0    1.9      MODIS         80       287.1  103.7   
5        310.2   4.0    1.9      MODIS         70       287.3   83.6   
10       300.4   1.0    1.0      MODIS         28       281.6    6.3   

                   HUC2 Name  \
1                Ohio Region   
3   Upper Mississippi Region   
4   Lower Mississippi Region   
5   Lower Mississippi Region   
10          Tennessee Region   

    Precipitation Accumulation (in) Start of Day Values  \
1                                                11.5     
3                                                 7.8     
4                                                11.3     
5                                                 8.4     
10                                               12.8     

    Precipitation Increment (in)  Air Temperature Average (degF)  \
1                            0.0                             2.0   
3                            0.0                            -8.0   
4                            0.0                            14.0   
5                            0.0                            11.0   
10                           0.0                            15.0   

    Soil Moisture Percent -2in (pct) Start of Day Values  \
1                                                36.1      
3                                                 4.9      
4                                                26.9      
5                                                 8.0      
10                                               33.1      

    Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  \
1                                78.0                       7.1   
3                                66.0                       1.8   
4                                79.0                      10.5   
5                                88.0                       4.2   
10                               88.0                       3.6   

    FIRE_DETECTED  1  Aqua  Terra  
1           False  0     0      1  
3           False  0     0      1  
4           False  0     0      1  
5           False  0     0      1  
10          False  0     1      0

In [21]:
#now one hot encode instrument
ohe = pd.get_dummies(dfMerged.instrument) #creates the one hot encoded df of instrument
dfMerged = pd.concat([dfMerged, ohe], axis=1) #merges the dataframe with the one hot encoded df
dfMerged = dfMerged.drop('instrument', axis = 1) #drop the instrument column
dfMerged.head()

brightness  scan  track confidence  bright_t31    frp  \
1        312.5   1.2    1.1         85       269.1   21.9   
3        309.8   4.0    1.9         68       289.9   82.0   
4        312.8   4.0    1.9         80       287.1  103.7   
5        310.2   4.0    1.9         70       287.3   83.6   
10       300.4   1.0    1.0         28       281.6    6.3   

                   HUC2 Name  \
1                Ohio Region   
3   Upper Mississippi Region   
4   Lower Mississippi Region   
5   Lower Mississippi Region   
10          Tennessee Region   

    Precipitation Accumulation (in) Start of Day Values  \
1                                                11.5     
3                                                 7.8     
4                                                11.3     
5                                                 8.4     
10                                               12.8     

    Precipitation Increment (in)  Air Temperature Average (degF)  \
1                            0.0                             2.0   
3                            0.0                            -8.0   
4                            0.0                            14.0   
5                            0.0                            11.0   
10                           0.0                            15.0   

    Soil Moisture Percent -2in (pct) Start of Day Values  \
1                                                36.1      
3                                                 4.9      
4                                                26.9      
5                                                 8.0      
10                                               33.1      

    Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  \
1                                78.0                       7.1   
3                                66.0                       1.8   
4                                79.0                      10.5   
5                                88.0                       4.2   
10                               88.0                       3.6   

    FIRE_DETECTED  1  Aqua  Terra  MODIS  VIIRS  
1           False  0     0      1      1      0  
3           False  0     0      1      1      0  
4           False  0     0      1      1      0  
5           False  0     0      1      1      0  
10          False  0     1      0      1      0

In [22]:
#the column name 'HUC2 Name' will not work for the necessary syntax because it has a space in it. 
dfMerged = dfMerged.rename(columns={"HUC2 Name": "HUC2Name"}) #renames the HUC2 Name column to remove spaces 

In [23]:
#now one hot encode the region by HUC2 name:
ohe = pd.get_dummies(dfMerged.HUC2Name) #creates the one hot encoded df of instrument
dfMerged = pd.concat([dfMerged, ohe], axis=1) #merges the dataframe with the one hot encoded df
dfMerged = dfMerged.drop('HUC2Name', axis = 1) #drop the instrument column
dfMerged.head()

brightness  scan  track confidence  bright_t31    frp  \
1        312.5   1.2    1.1         85       269.1   21.9   
3        309.8   4.0    1.9         68       289.9   82.0   
4        312.8   4.0    1.9         80       287.1  103.7   
5        310.2   4.0    1.9         70       287.3   83.6   
10       300.4   1.0    1.0         28       281.6    6.3   

    Precipitation Accumulation (in) Start of Day Values  \
1                                                11.5     
3                                                 7.8     
4                                                11.3     
5                                                 8.4     
10                                               12.8     

    Precipitation Increment (in)  Air Temperature Average (degF)  \
1                            0.0                             2.0   
3                            0.0                            -8.0   
4                            0.0                            14.0   
5                            0.0                            11.0   
10                           0.0                            15.0   

    Soil Moisture Percent -2in (pct) Start of Day Values  \
1                                                36.1      
3                                                 4.9      
4                                                26.9      
5                                                 8.0      
10                                               33.1      

    Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  \
1                                78.0                       7.1   
3                                66.0                       1.8   
4                                79.0                      10.5   
5                                88.0                       4.2   
10                               88.0                       3.6   

    FIRE_DETECTED  1  Aqua  Terra  MODIS  VIIRS  Arkansas-White-Red Region  \
1           False  0     0      1      1      0                          0   
3           False  0     0      1      1      0                          0   
4           False  0     0      1      1      0                          0   
5           False  0     0      1      1      0                          0   
10          False  0     1      0      1      0                          0   

    California Region  Great Basin Region  Great Lakes Region  Hawaii Region  \
1                   0                   0                   0              0   
3                   0                   0                   0              0   
4                   0                   0                   0              0   
5                   0                   0                   0              0   
10                  0                   0                   0              0   

    Lower Colorado Region  Lower Mississippi Region  Mid Atlantic Region  \
1                       0                         0                    0   
3                       0                         0                    0   
4                       0                         1                    0   
5                       0                         1                    0   
10                      0                         0                    0   

    Missouri Region  New England Region  Ohio Region  \
1                 0                   0            1   
3                 0                   0            0   
4                 0                   0            0   
5                 0                   0            0   
10                0                   0            0   

    Pacific Northwest Region  Rio Grande Region  Souris-Red-Rainy Region  \
1                          0                  0                        0   
3                          0                  0                        0   
4                          0                  0                        0   
5                          0                  0              

In [24]:
#check the data types to make sure they are all values and not strings
dfMerged.dtypes

brightness                                              float64
scan                                                    float64
track                                                   float64
confidence                                               object
bright_t31                                              float64
frp                                                     float64
Precipitation Accumulation (in) Start of Day Values     float64
Precipitation Increment (in)                            float64
Air Temperature Average (degF)                          float64
Soil Moisture Percent -2in (pct) Start of Day Values    float64
Relative Humidity Enclosure (pct)                       float64
Wind Speed Average (mph)                                float64
FIRE_DETECTED                                              bool
1                                                         uint8
Aqua                                                      uint8
Terra                                   

In [25]:
#confidence appears to be a string. check how many are not numeric
dfMerged[pd.to_numeric(dfMerged['confidence'], errors='coerce').notnull()].shape

(424973, 37)

In [26]:
#confidence is not numeric enough. drop it for now in this initial ML approach
#n might equate to 0 and should be researched further 
dfMerged = dfMerged.drop('confidence', axis=1)
dfMerged.head()

brightness  scan  track  bright_t31    frp  \
1        312.5   1.2    1.1       269.1   21.9   
3        309.8   4.0    1.9       289.9   82.0   
4        312.8   4.0    1.9       287.1  103.7   
5        310.2   4.0    1.9       287.3   83.6   
10       300.4   1.0    1.0       281.6    6.3   

    Precipitation Accumulation (in) Start of Day Values  \
1                                                11.5     
3                                                 7.8     
4                                                11.3     
5                                                 8.4     
10                                               12.8     

    Precipitation Increment (in)  Air Temperature Average (degF)  \
1                            0.0                             2.0   
3                            0.0                            -8.0   
4                            0.0                            14.0   
5                            0.0                            11.0   
10                           0.0                            15.0   

    Soil Moisture Percent -2in (pct) Start of Day Values  \
1                                                36.1      
3                                                 4.9      
4                                                26.9      
5                                                 8.0      
10                                               33.1      

    Relative Humidity Enclosure (pct)  Wind Speed Average (mph)  \
1                                78.0                       7.1   
3                                66.0                       1.8   
4                                79.0                      10.5   
5                                88.0                       4.2   
10                               88.0                       3.6   

    FIRE_DETECTED  1  Aqua  Terra  MODIS  VIIRS  Arkansas-White-Red Region  \
1           False  0     0      1      1      0                          0   
3           False  0     0      1      1      0                          0   
4           False  0     0      1      1      0                          0   
5           False  0     0      1      1      0                          0   
10          False  0     1      0      1      0                          0   

    California Region  Great Basin Region  Great Lakes Region  Hawaii Region  \
1                   0                   0                   0              0   
3                   0                   0                   0              0   
4                   0                   0                   0              0   
5                   0                   0                   0              0   
10                  0                   0                   0              0   

    Lower Colorado Region  Lower Mississippi Region  Mid Atlantic Region  \
1                       0                         0                    0   
3                       0                         0                    0   
4                       0                         1                    0   
5                       0                         1                    0   
10                      0                         0                    0   

    Missouri Region  New England Region  Ohio Region  \
1                 0                   0            1   
3                 0                   0            0   
4                 0                   0            0   
5                 0                   0            0   
10                0                   0            0   

    Pacific Northwest Region  Rio Grande Region  Souris-Red-Rainy Region  \
1                          0                  0                        0   
3                          0                  0                        0   
4                          0                  0                        0   
5                          0                  0                        0   
10                         0                  0    

The DF should be ready for ml

## Begin ML Preperation  

In [27]:
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from sklearn import svm
from sklearn.neural_network import MLPClassifier
from sklearn.preprocessing import StandardScaler, LabelEncoder
from sklearn.metrics import confusion_matrix, classification_report
from sklearn.model_selection import train_test_split
%matplotlib inline 

In [28]:
#seperate data sets as labels and features
X = dfMerged.drop('FIRE_DETECTED', axis=1)
y = dfMerged['FIRE_DETECTED']

In [29]:
#train test splitting of data
#common syntax here is to use X_train, X_test, y_train, y_test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42) #random state is a random seed

In [30]:
#create our scalar to get optimized result
sc = StandardScaler() #runs the standard scalar with default settings. you can refine this, see docs
#transform the feature data by using the fit_transform 
X_train = sc.fit_transform(X_train) 
X_test = sc.fit_transform(X_test) 
X_train[:5] #take a look at the standardized data

array([[ 0.13552914, -0.3585452 , -0.70829839,  0.35364811, -0.18887688,
         0.43959927, -0.16793426, -0.19530779, -0.34143051,  0.77302252,
        -0.96838203,  0.66224786, -0.43018942, -0.41794577, -0.66224786,
         0.66224786, -0.24381111, -0.79494974, -0.20312761, -0.0804431 ,
        -0.08345914, -0.1853519 , -0.14635327, -0.11674081, -0.24955193,
        -0.01409693, -0.14017897, -0.31199502, -0.0654225 , -0.05540131,
         2.93226398, -0.07953652, -0.29742034, -0.17026326, -0.20942564],
       [-0.28175646,  3.49381565,  2.57065382,  0.73594194,  0.19655327,
        -0.70964052, -0.16793426,  0.60780905, -0.35512165, -0.5411481 ,
         0.13463026, -1.51000865,  2.32455737, -0.41794577,  1.51000865,
        -1.51000865, -0.24381111,  1.25794117, -0.20312761, -0.0804431 ,
        -0.08345914, -0.1853519 , -0.14635327, -0.11674081, -0.24955193,
        -0.01409693, -0.14017897, -0.31199502, -0.0654225 , -0.05540131,
        -0.34103342, -0.07953652, -0.29742034, -0.

## Begin running ML Algorithms

#### Random Forest Classifier 200 estimators

In [31]:
testmodel = RandomForestClassifier(n_estimators=200) #n estimators is hyperparameter how many trees are in the forest
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now
pred[:20] #take a look at the predictions

array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False])

In [33]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.98      1.00      0.99    271652
        True       0.90      0.26      0.40      7687

    accuracy                           0.98    279339
   macro avg       0.94      0.63      0.70    279339
weighted avg       0.98      0.98      0.97    279339

Confusion matrix:
[[271443    209]
 [  5707   1980]]


#### Random Forest Classifier 100 estimators

In [34]:
testmodel = RandomForestClassifier(n_estimators=100) #n estimators is hyperparameter how many trees are in the forest
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now
pred[:20] #take a look at the predictions

array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False])

In [35]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.98      1.00      0.99    271652
        True       0.89      0.26      0.40      7687

    accuracy                           0.98    279339
   macro avg       0.93      0.63      0.70    279339
weighted avg       0.98      0.98      0.97    279339

Confusion matrix:
[[271397    255]
 [  5688   1999]]


#### Neural Network

In [36]:
testmodel = MLPClassifier(hidden_layer_sizes=(11, 11, 11), max_iter=1000) #these are mandatory hyperparameters for neural networks
testmodel.fit(X_train, y_train)
pred = testmodel.predict(X_test)

In [37]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.97      1.00      0.99    271652
        True       0.54      0.02      0.03      7687

    accuracy                           0.97    279339
   macro avg       0.76      0.51      0.51    279339
weighted avg       0.96      0.97      0.96    279339

Confusion matrix:
[[271553     99]
 [  7571    116]]


#### Logistic Regression

In [39]:
#import logistic regression import things
from sklearn.linear_model import LogisticRegressionCV, LogisticRegression, SGDClassifier

In [40]:
testmodel = LogisticRegression(solver='lbfgs')
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now

In [41]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.97      1.00      0.99    271652
        True       0.12      0.00      0.00      7687

    accuracy                           0.97    279339
   macro avg       0.55      0.50      0.49    279339
weighted avg       0.95      0.97      0.96    279339

Confusion matrix:
[[271637     15]
 [  7685      2]]


#### Logistic regression CV

In [42]:
testmodel = LogisticRegressionCV(cv=3)
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now

In [43]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.97      1.00      0.99    271652
        True       0.00      0.00      0.00      7687

    accuracy                           0.97    279339
   macro avg       0.49      0.50      0.49    279339
weighted avg       0.95      0.97      0.96    279339

Confusion matrix:
[[271649      3]
 [  7687      0]]


#### SGD Classifier

In [44]:
testmodel = SGDClassifier(max_iter=100, tol=1e-3)
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now

In [45]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.97      1.00      0.99    271652
        True       0.00      0.00      0.00      7687

    accuracy                           0.97    279339
   macro avg       0.49      0.50      0.49    279339
weighted avg       0.95      0.97      0.96    279339

Confusion matrix:
[[271651      1]
 [  7687      0]]


#### Extra Trees Classifier

In [46]:
from sklearn.ensemble import BaggingClassifier, ExtraTreesClassifier

In [47]:
testmodel = ExtraTreesClassifier(n_estimators=100)
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now

In [48]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.98      1.00      0.99    271652
        True       0.92      0.26      0.41      7687

    accuracy                           0.98    279339
   macro avg       0.95      0.63      0.70    279339
weighted avg       0.98      0.98      0.97    279339

Confusion matrix:
[[271469    183]
 [  5684   2003]]


#### Bagging Classifier

In [49]:
testmodel = BaggingClassifier()
testmodel.fit(X_train, y_train) #fits the model using training data
pred = testmodel.predict(X_test) #predict the test data now

In [50]:
#see how the model peformed
print('Report:')
print(classification_report(y_test, pred))
print('Confusion matrix:')
print(confusion_matrix(y_test, pred))

Report:
              precision    recall  f1-score   support

       False       0.98      0.99      0.99    271652
        True       0.57      0.31      0.40      7687

    accuracy                           0.97    279339
   macro avg       0.78      0.65      0.69    279339
weighted avg       0.97      0.97      0.97    279339

Confusion matrix:
[[269895   1757]
 [  5331   2356]]
